src/integrationTest/java/org/opensearch/security/grpc/GrpcHelpers.java

```diff
@@ -0,0 +1,158 @@
+        TEST_CERTIFICATES.getNodeKey(0, null).getAbsolutePath(),
+        SECURITY_SSL_AUX_PEMCERT_FILEPATH.getConcreteSettingForNamespace(GRPC_SECURE_TRANSPORT_SETTING_KEY).getKey(),
+        TEST_CERTIFICATES.getNodeCertificate(0).getAbsolutePath(),
+        SECURITY_SSL_AUX_PEMTRUSTEDCAS_FILEPATH.getConcreteSettingForNamespace(GRPC_SECURE_TRANSPORT_SETTING_KEY).getKey(),
+        TEST_CERTIFICATES.getRootCertificate().getAbsolutePath()
+    );
+
+    public static LocalCluster.Builder baseGrpcCluster() {
+        return new LocalCluster.Builder().clusterManager(ClusterManager.SINGLENODE)
+            .plugin(GrpcPlugin.class)
+            .certificates(TEST_CERTIFICATES)
+            .nodeSettings(SECURE_GRPC_TRANSPORT_SETTINGS)
+            .loadConfigurationIntoIndex(false)
+            .sslOnly(true);
+    }
+
+    /*
+    Plaintext connection.
+    No encryption in transit.
+    */
+    public static ManagedChannel plaintextChannel() {
+        return NettyChannelBuilder.forAddress(HOST_ADDR, PORTS_RANGE.ports()[0]).proxyDetector(NOOP_PROXY_DETECTOR).usePlaintext().build();
+    }
+
+    /*
+    TLS with no client certificate.
+    */
+    public static ManagedChannel insecureChannel() {
+        ChannelCredentials credentials = TlsChannelCredentials.newBuilder()
+            .trustManager(InsecureTrustManagerFactory.INSTANCE.getTrustManagers())
+            .build();
+        return Grpc.newChannelBuilderForAddress(HOST_ADDR, PORTS_RANGE.ports()[0], credentials).build();
+    }
+
+    /*
+    TLS with client certificate trusted by server.
+    */
+    public static ManagedChannel secureChannel() throws IOException {
+        ChannelCredentials credentials = TlsChannelCredentials.newBuilder()
+            .keyManager(TEST_CERTIFICATES.getNodeCertificate(0), TEST_CERTIFICATES.getNodeKey(0, null))
+            .trustManager(InsecureTrustManagerFactory.INSTANCE.getTrustManagers())
+            .build();
+        return Grpc.newChannelBuilderForAddress(HOST_ADDR, PORTS_RANGE.ports()[0], credentials).build();
+    }
+
+    /*
+    TLS with client certificate not trusted by server.
+    */
+    public static ManagedChannel secureUntrustedChannel() throws IOException {
+        ChannelCredentials credentials = TlsChannelCredentials.newBuilder()
+            .keyManager(UN_TRUSTED_TEST_CERTIFICATES.getNodeCertificate(0), UN_TRUSTED_TEST_CERTIFICATES.getNodeKey(0, null))
+            .trustManager(InsecureTrustManagerFactory.INSTANCE.getTrustManagers())
+            .build();
+        return Grpc.newChannelBuilderForAddress(HOST_ADDR, PORTS_RANGE.ports()[0], credentials).build();
+    }
+
+    public static BulkResponse doBulk(ManagedChannel channel, String index, long numDocs) {
+        BulkRequest.Builder requestBuilder = BulkRequest.newBuilder().setRefresh(Refresh.REFRESH_TRUE);
+        for (int i = 0; i < numDocs; i++) {
+            String docBody = "{\"field\": \"doc " + i + " body\"}";
```

Comment:
nit: prefer to use TextBlocks for JSON payloads now that java min compatibility is 21